demos/https/iot_demo_https_sync.c

```diff
@@ -0,0 +1,499 @@
+        if( httpsClientStatus != IOT_HTTPS_OK )
+        {
+            IotLogError( "Failed to write the header Range: %.*s into the request. With error code: %d", numWritten, rangeValueStr, httpsClientStatus );
+            IOT_SET_AND_GOTO_CLEANUP( EXIT_FAILURE );
+        }
+
+        /* Send the request and receive the response synchronously. */
+        IotLogInfo("Now requesting Range: %s.", rangeValueStr);
+        /* A new response handle is returned from IotHttpsClient_SendSync(). We reuse the respHandle variable because
+           the last response was already processed fully.  */
+
+        httpsClientStatus = IotHttpsClient_SendSync( &connHandle, reqHandle, &respHandle );
+        if( httpsClientStatus != IOT_HTTPS_OK )
+        {
+            IotLogError( "There has been an error receiving the response. The error code is: %d", httpsClientStatus );
+            IOT_SET_AND_GOTO_CLEANUP( EXIT_FAILURE );
+        }
+
+        httpsClientStatus = IotHttpsClient_ReadResponseStatus(respHandle, &respStatus );
+        if( httpsClientStatus != IOT_HTTPS_OK )
+        {
+            IotLogError("Error in retreiving the response status. Error code %d", httpsClientStatus);
+            IOT_SET_AND_GOTO_CLEANUP( EXIT_FAILURE );
+        }
+        
+        if( respStatus != IOT_HTTPS_STATUS_PARTIAL_CONTENT)
+        {
+            IotLogError("Failed to retrieve the partial content response from s3. Response status: %d", respStatus);
+            IOT_SET_AND_GOTO_CLEANUP( EXIT_FAILURE );
+        }
+
+        /* Get the content length of the body for printing without a NULL terminator. */
+        httpsClientStatus = IotHttpsClient_ReadContentLength( respHandle, &contentLength );
+        if( httpsClientStatus != IOT_HTTPS_OK )
+        {
+            IotLogError("Failed to read the Content-Length from the response. Error code %d", httpsClientStatus);
+            IOT_SET_AND_GOTO_CLEANUP( EXIT_FAILURE );
+        }
+
+        /* The response has been fully received. */
+        IotLogInfo( "Response return code: %d", respStatus );
+        /* The logging buffer may not fit all of the response body received and the output on the console will be truncated
+           to the first configLOGGING_MAX_MESSAGE_LENGTH number of characters. */
+        IotLogInfo( "Response Body: \r\n%.*s", contentLength, _pRespBodyBuffer );
+
+        /* We increment by the contentLength because the server may not have sent us the range we request. */
+        curByte += contentLength;
+
+        IotLogInfo("Downloaded %d/%d", curByte, fileSize);
+
+        /* If amount of file remaining to request is less than the current amount of bytes to request next time, then
+           udpdate the amount of bytes to request, on the next iteration, to be the amount remaining. */
+        if( (fileSize - curByte) < numReqBytes)
+        {
+            numReqBytes = fileSize - curByte;
+        }
```

Comment:
There are not 10 connections established to the server. because:
```
 * If parameter pConnHandle is passed in pointing to a NULL #IotHttpsConnectionHandle_t, then a connection is to be made 
 * implicitly and a valid connection handle in pConnHandle will be returned. 
 * If parameter pConnHandle points to a non-NULL #IotHttpsConnectionHandle_t and it is in a disconnected state, then a 
 * connection is to be made implicitly and valid connection handle in pConnHandle will be returned. 
 * In each implicit connection case #IotHttpsRequestHandle_t reqHandle must have been returned from 
 * @ref https_client_function_initializerequest where the #IotHttpsRequestInfo_t request configuration contained a 
 * non-NULL #IotHttpsRequestInfo_t.pConnInfo. Please see #IotHttpsConnectionInfo_t for connection configurations. 
```

Reply:
State is set internally that the connection is active and can only be inactive when the disconnect API is called.